### PR TITLE
Benchmark + optimize

### PR DESCRIPTION
**Pull request by samouri**
## summary
Creates a super basic `benchmark.mjs` file for measuring `createElement` performance.
Then makes 4 small changes that each remove memory leaks and/or improve ctor performance 

**initial performance**
```
createElement x 197,202 ops/sec ±5.21% (72 runs sampled)
```

**opt: skip HTMLElement ctor store**
```
createElement x 449,453 ops/sec ±2.38% (83 runs sampled)
```
**opt: getter for CSSStyleDeclaration (lazy instantiation)**
```
createElement x 626,866 ops/sec ±2.26% (85 runs sampled)
```

**opt: class field loose transform**
```
createElement x 643,255 ops/sec ±2.16% (86 runs sampled)
```

**opt: remove minification step** (note: I don't know why this has faster output)
```
createElement x 726,293 ops/sec ±0.77% (91 runs sampled)
```

---

cc @rcebulko

### CODE REVIEW

File: src/main-thread/main-thread.d.ts [hunk 1]
```diff
@@ -62,3 +62,4 @@ type RenderableElement = (HTMLElement | SVGElement | Text | Comment) & {
 
 declare const WORKER_DOM_DEBUG: boolean;
 declare const IS_AMP: boolean;
+declare const IS_SERVER: boolean;
```
Comment (rcebulko) [inline src/main-thread/main-thread.d.ts:65]: 🚲 🏚️ : 
```suggestion
declare const DISABLE_TRANSFERS: boolean;
```

Comment (samouri) [reply src/main-thread/main-thread.d.ts]: WDYT of the new thing I did, which is kind of nice IMO. Attached to `process.env.SERVER`

**Review by rcebulko — APPROVED** [review]

File: config/rollup.main-thread.js [hunk 2]
```diff
@@ -38,13 +31,7 @@ const ESModules = [
...
     plugins: [
       removeWorkerWhitespace(),
-      replace({
-        values: {
-          WORKER_DOM_DEBUG: true,
-          IS_AMP: false,
-        },
-        preventAssignment: true,
-      }),
+      replacePlugin({ debug: true }),
```
[the remark this answers is not in the document]
Comment (samouri) [reply config/rollup.main-thread.js:34]: This is configuration for the output file: `dist/debug/main.mjs` (scroll up a few lines to see that).
For this binary we want all of the debug code. Very similar to amphtml's `mode`

File: src/worker-thread/dom/Element.ts [hunk 3]
```diff
@@ -100,6 +100,13 @@ export class Element extends ParentNode {
     ];
   }
 
+  get style(): CSSStyleDeclaration {
```
[the remark this answers is not in the document]
Comment (samouri) [reply src/worker-thread/dom/Element.ts:103]: Great idea! I'll make this change

File: benchmark.mjs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+var suite = new Benchmark.Suite();
+suite
+  .add('createElement', function () {
+    doc.createElement('test-elem');
+  })
+  .on('complete', function () {
+    const results = Array.from(this);
+    console.log(results.join('\n'));
+  })
+  .run();
```
[the remark this answers is not in the document]
Comment (samouri) [reply benchmark.mjs:15]: ignore complete

**Pull request merged**